src/backend/distributed/commands/utility_hook.c

```diff
@@ -632,6 +637,32 @@ citus_ProcessUtilityInternal(PlannedStmt *pstmt,
 		{
 			ddlJobs = ops->preprocess(parsetree, queryString, context);
 		}
+
+		if (ddlJobs != NIL)
```

Comment:
I guess this isn't a completely generic solution at this point, because it does not work when postprocess returns the ddljobs. So for DDL where postprocess is the one that's necessary we'd need some other solution (e.g. only taking the locks on the coordinator and not executing the SQL yet). I think this PR seems like a good first step though, and the postprocess one could be solved in a follow up PR.

Reply:
> I guess this isn't a completely generic solution at this point, because it does not work when postprocess returns the ddljobs.

Correct.

> So for DDL where postprocess is the one that's necessary we'd need some other solution (e.g. only taking the locks on the coordinator and not executing the SQL yet). 

Right, we either need this^ or need to move Postprocess pieces to Preprocess stage for the commands we want to support from any-node.